core/src/main/java/org/hisp/dhis/android/core/fileresource/internal/FileResourceDownloadCall.kt

```diff
@@ -169,25 +170,32 @@ internal class FileResourceDownloadCall @Inject constructor(
         }
     }
 
-    private fun <V> downloadFile(
+    @Suppress("NestedBlockDepth")
+    private suspend fun <V> downloadFile(
         value: V,
         maxContentLength: Int?,
-        download: (V) -> Call<ResponseBody>?,
+        download: suspend (V) -> ResponseBody?,
```

Comment:
This function, couldn't it be NonNull? like `download: suspend (V) -> ResponseBody`